misc/phase_retrieval.ipynb

```diff
@@ -167,31 +292,73 @@
    "source": [
     "fig, axs = plt.subplots(1,2, figsize=(12,5))\n",
     "ax = axs[0]\n",
-    "ax.plot(recon.array[67,70,50:90])\n",
-    "ax.plot(recon_attenuation.array[67,70,50:90])\n",
-    "ax.plot(recon_filter.array[67,70,50:90])\n",
-    "ax.plot(recon_tomopy.array[67,70,50:90],'--')\n",
-    "ax.plot(recon_tomopy_abs.array[67,70,50:90],'--')\n",
+    "vertical_slice = 67\n",
+    "y_slice = 70\n",
+    "x_range = range(50,90)\n",
+    "ax.plot(x_range, recon.array[vertical_slice,y_slice,x_range])\n",
+    "ax.plot(x_range, recon_attenuation.array[vertical_slice,y_slice,x_range])\n",
+    "ax.plot(x_range, recon_filter.array[vertical_slice,y_slice,x_range])\n",
+    "ax.plot(x_range, recon_tomopy.array[vertical_slice,y_slice,x_range],'--')\n",
+    "ax.plot(x_range, recon_tomopy_abs.array[vertical_slice,y_slice,x_range],'--')\n",
     "\n",
     "ax.set_xlabel('Horizontal x')\n",
     "ax.set_ylabel('Intensity')\n",
-    "ax.legend(['Original image', 'Phase retrieval (scaled)', 'Phase retrieval - full_retrieval=False', 'Tomopy transmission', 'Tomopy absorption'])\n",
+    "ax.set_title('Line Profile at horizontal_y=' + str(y_slice) + ', vertical slice=' + str(vertical_slice))\n",
```

Comment:
It looks like the line profiles have been done in regions outside of the cropped region displayed in the figure above. Is there a particular reason for this choice? I think it would be nice to have a line profile in a region that we have displayed with show2D as well

Reply:
Hi Laura, thank you for reviewing again! I think the slice plot is from the cropped region, just that because I'm plotting a subset of the data `show2D(data.array[67, [50:90],[50:90])` the axes show from 0 to 40 rather than 50 to 90. I don't know if there's a way to use data in show2D with a subset, I don't think I can make `slice_list` take a range. I've added a plot showing the full data range so hopefully now it's clearer that the slice is in the subset.